### PR TITLE
[Intl] Document the emoji transliterator

### LINKED ISSUE #17069: [Intl] Add `EmojiTransliterator` to translate emoji to many locales
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#46755
| PR author(s) | @lyrixx, @nicolas-grekas
| Merged in    | 6.2

### PR DESCRIPTION
**Pull request by javiereguiluz**
Fixes #17069.

### CODE REVIEW

File: components/intl.rst [hunk 1]
```diff
@@ -359,6 +360,31 @@ to catching the exception, you can also check if a given timezone ID is valid::
 
     $isValidTimezone = Timezones::exists($timezoneId);
 
+.. _component-intl-emoji-transliteration:
+
+Emoji Transliteration
+~~~~~~~~~~~~~~~~~~~~~
+
+The ``EmojiTransliterator`` class provides an utility to translate emojis into
+their textual represtation in all languages::
```
Comment (wouterj) [inline components/intl.rst]: "... based on the [Unicode CLDR dataset](https://github.com/unicode-org/cldr)"?

**Review by wouterj — APPROVED** [review]

File: components/intl.rst [hunk 2]
```diff
@@ -359,6 +360,31 @@ to catching the exception, you can also check if a given timezone ID is valid::
...
+
+    // describe emojis in Ukrainian
+    $transliterator = EmojiTransliterator::create('uk');
+    $transliterator->transliterate('Menus with 🍕 or 🍝');
+    // => 'Menus with піца or спагеті'
+
+.. tip::
+
+    Combine this emoji transliterator with the :ref:`Symfony String slugger <string-slugger>`
+    to impove the slugs of contents that include emojis (e.g. for URLs).
```
Comment (OskarStark) [inline components/intl.rst]: Improve

File: components/intl.rst [hunk 3]
```diff
@@ -359,6 +360,31 @@ to catching the exception, you can also check if a given timezone ID is valid::
 
     $isValidTimezone = Timezones::exists($timezoneId);
 
+.. _component-intl-emoji-transliteration:
+
+Emoji Transliteration
+~~~~~~~~~~~~~~~~~~~~~
```
Comment (OskarStark) [inline components/intl.rst:366]: Versionadded?

**Comment by javiereguiluz** [comment]
Thanks for the reviews! I've done all the suggested changes.

**Comment by lyrixx** [comment]
Thanks @javiereguiluz

File: components/intl.rst [hunk 4]
```diff
@@ -359,6 +360,35 @@ to catching the exception, you can also check if a given timezone ID is valid::
...
+
+.. versionadded:: 6.2
+
+    The Empoji transliteration feature was introduced in Symfony 6.2.
```
Comment (qdequippe) [inline components/intl.rst:370]: Just a little typo here :) "Emoji"

Comment (xabbuh) [reply components/intl.rst:370]: good catch :) fixed in 863dcbe15

**Pull request merged**